app/controllers/admin/quotes_controller.rb

```diff
@@ -0,0 +1,26 @@
+class Admin::QuotesController < ApplicationController
+  before_action :authenticate_user
+  before_action :set_quote, only: [:approve]
+
+  def pending
+    @pagy, @quotes = pagy(@quotes = Quote.where(state: 'pending'))
+  end
+
+  def approve
+    @quote.approve!
+
+    redirect_to admin_pending_quotes_path
+  end
+
+  private
+
+  def set_quote
+    @quote ||= Quote.find(params[:quote_id])
+  end
+
+  def authenticate_user
+    return if current_user && (current_user.admin? || current_user.moderator?)
```

Comment:
don't check presence of `current_user`, add `before_action :authenticate_user!` instead (the method from devise)